media-source-respec.html

```diff
@@ -2719,6 +2711,280 @@ <h3>Event Summary</h3>
+      </p>
+      <ol class="algorithm">
+        <li>Run the {{MediaSource}} [=MediaSource/SourceBuffer Monitoring=] algorithm.</li>
+        <li>Let |can play uninterupted and efficiently| be a flag that is true if the {{HTMLMediaElement/buffered}} attribute contains a {{TimeRanges}} that includes the current playback position and [=enough managed data to ensure uninterrupted playback=]
+          and is [=able to retrieve and buffer data in an efficient way=]
+          <dl class="switch">
+            <dt>If |can play uninterupted and efficiently| is not equal to {{ManagedMediaSource/streaming}}, [=queue an element task=] on the [=media element=]
+              that runs the following steps:
+            </dt>
+            <dd>
+              <ol>
+                <li>Set |this| {{ManagedMediaSource/streaming}} attribute to |can play uninterupted and efficiently|.</li>
+                <li>If |can play uninterupted and efficiently| is false [=fire an event=] called {{startstreaming}} at the {{ManagedMediaSource}}.</li>
+                <li>Else [=fire an event=] called {{endstreaming}} at the {{ManagedMediaSource}}.</li>
+              </ol>
+            </dd>
+          </dl>
+        </li>
+      </ol>
+
+      <h4><dfn data-dfn-for="ManagedMediaSource">Memory Cleanup</dfn></h4>
+      <ol>
+        <li>
+          <dl class="switch">
+            <dt>
+              For each |buffer:ManagedSourceBuffer| in [=this=]'s
+              {{MediaSource/sourceBuffers}}:
+            </dt>
+            <dd>
+              <ol>
+                <li>Run the |buffer|'s [=ManagedSourceBuffer/memory cleanup=] algorithm.</li>
+              </ol>
+            </dd>
+          </dl>
+        </li>
+      </ol>
+    </section>
+
+    <section data-dfn-for="BufferedChangeEvent">
+      <h2><dfn>BufferedChangeEvent</dfn> interface</h2>
+      <pre class="idl">
+        [Exposed=(Window,DedicatedWorker)]
+        interface BufferedChangeEvent : Event {
+          constructor(DOMString type, BufferedChangeEventInit eventInitDict);
+
+          [SameObject] readonly attribute TimeRanges addedRanges;
+          [SameObject] readonly attribute TimeRanges removedRanges;
+        };
+
+        dictionary BufferedChangeEventInit : EventInit {
+          TimeRanges addedRanges;
+          TimeRanges removedRanges;
+        };
+      </pre>
+      <h3>Attributes</h3>
+      <dl>
+        <dt>
+          <dfn>addedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges added during the last run of the [=coded frame processing=] algorithm.
+        </dd>
+        <dt>
+          <dfn>removedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges removed during the last run of the [=coded frame removal=] or [=coded frame eviction=] algorithm or if the
+          user agent evicted content in response to a [=ManagedSourceBuffer/memory cleanup=].
```

Comment:
nit: "during the last run" is slightly confusing. [SameObject] implies "the last run" changes (which it should), but "during the last run" could be misinterpreted as whatever the most recent of these runs is whenever those attributes are queried.

Reply:
The attribute is not queried, but represents the state of the coalesced changes that occurred during the last completed operation (between updatestart and updateend... and whatever was removed if there was memory pressure, which can't be distinguished anyway).